### ISSUE TITLE
"tsc-delayed-rebuild" message causing extra rebuilds

### ISSUE THREAD

**Issue by csantero** [typed-ember/ember-cli-typescript#191]
Testing with 1.3.0-beta.1 I notice whenever I change a .ts file I sometimes (not always) get an additional rebuild step, which is adding several seconds to incremental change cycles.

In the following example, the first time I changed a .ts file, it didn't touch the `tsc-delayed-rebuild` file, but the second time, it did:

```
$ ember serve
Could not start watchman
Visit https://ember-cli.com/user-guide/#watchman for more info.
Livereload server on http://localhost:37729

Build successful (93523ms) – Serving on http://localhost:6200/

file changed routes/application.ts

Build successful (3293ms) – Serving on http://localhost:6200/



Slowest Nodes (totalTime => 5% )              | Total (avg)         
----------------------------------------------+---------------------
GraphQLFilter (2)                             | 847ms (423 ms)      
SourceMapConcat: Concat: App (1)              | 638ms               
broccoli-persistent-filter:Babel > [Ba... (3) | 310ms (103 ms)      
TypescriptOutput (3)                          | 251ms (83 ms)       
SourceMapConcat: Concat: Vendor /asset... (1) | 217ms               

file changed routes/application.ts
file changed tsc-delayed-rebuild

Build successful (2679ms) – Serving on http://localhost:6200/



Slowest Nodes (totalTime => 5% )              | Total (avg)         
----------------------------------------------+---------------------
SourceMapConcat: Concat: App (1)              | 700ms               
GraphQLFilter (2)                             | 681ms (340 ms)      
SourceMapConcat: Concat: Vendor /asset... (1) | 284ms               


Build successful (2193ms) – Serving on http://localhost:6200/



Slowest Nodes (totalTime => 5% )              | Total (avg)         
----------------------------------------------+---------------------
GraphQLFilter (2)                             | 752ms (376 ms)      
TypescriptOutput (3)                          | 250ms (83 ms)       
SourceMapConcat: Concat: Vendor /asset... (1) | 242ms               
SimpleConcatConcat (15)                       | 119ms (7 ms)   
```

**Comment by dfreeman** [comment]
The way ember-cli-typescript works, any time a file changes we essentially have a race between the TS compiler and Broccoli/Ember CLI to pick up on that change. If the Broccoli file watcher notices a change and `tsc` hasn't yet, it can mean one of two things: either something other than a `.ts` file has changed, and we should just continue forward with the previous output from `tsc`, or the TypeScript watcher just hasn't noticed the change yet, and we should wait for it to rebuild.

Given this setup, we can't know in any given instance which of those two scenarios we're in, so we impose a maximum wait time for `tsc` to indicate that it's starting a rebuild before we just give up and move on. That `tsc-delayed-rebuild` file is essentially a failsafe put in place to handle it if we ever move on too soon, and it turns out tsc needed to rebuild after all and was just slow — it's a dummy file that we touch when we need to trigger a Broccoli rebuild after the fact.

When the work in #172 landed, I shortened the default threshold from 500ms to 250ms under the assumption that fixing a lot of our file watcher thrashing issues would also lessen the typical delay between the two watchers noticing a change, but it looks like that may have been premature. @csantero if you try tweaking [this line](https://github.com/typed-ember/ember-cli-typescript/blob/c7d4f923e3076bcc35e5fb511396a985dfb0cca4/lib/incremental-typescript-compiler.js#L33) in your local copy of ember-cli-typescript back to `500`, can you see if that makes your problem go away? If so, we should likely a) put that back and b) possibly make it configurable on a per-project basis.

**Comment by dfreeman** [comment]
(On an unrelated note: if you upgrade to the latest `ember-apollo-client@1.0.1`, you should see `GraphQLFilter` completely drop out of that 'Slowest Nodes' list 🙂)

**Comment by csantero** [comment]
I tried this with an `autoresolveThreshold` of up to 2500 and it still happens.

**Comment by dfreeman** [comment]
Thanks @csantero, that's super helpful to know! I'll try to put together a build with beefed up debug logging sometime in the next couple of days—I think to try and pin this down we're just going to need to gather a lot of information to paint a picture of the order of events on your machine.

**Comment by dfreeman** [comment]
Should be fixed in `1.3.0-beta.2` 🎉
